src/storage/mongodb-store.ts

```diff
@@ -0,0 +1,111 @@
+import { injectable } from 'inversify';
+import { ConfigManager } from '../config';
+import mongoose from 'mongoose';
+import { Item } from '../entity/Item';
+import { ItemType } from '../entity/item-type';
+import { Publisher } from '../entity/publisher';
+import { Team } from '../entity/Team';
+import { PersistentStorage } from '../types';
+
+@injectable()
+export class MongodbStore<T> implements PersistentStorage<T> {
+    private _itemModel: mongoose.Model<mongoose.Document & Item<number>>;
+    public deleteItem(id: T): Promise<boolean> {
+        return undefined;
+    }
+
+    public getItem(id: T): Promise<Item<T> | null> {
+        return undefined;
+    }
+
+    public hasItem(id: T): Promise<boolean> {
+        return undefined;
+    }
+
+    public async filterItemNotStored(ids: T[]): Promise<T[]> {
+        const checkIds = ids.map(id => this._findById(id));
+        return Promise.all(checkIds).then(vals => {
+            const notStoredIds: T[] = [];
+            vals.forEach((val, index) => {
+                if (!val) {
+                    notStoredIds.push(ids[index])
+                }
+            })
+            return notStoredIds;
+        })
+    }
+
+    public async putItem(item: Item<T>): Promise<boolean> {
+        const itemData = new this._itemModel(item);
+        itemData.save(function (err) {
+            if (!err) console.log('Success!');
+        });
+        return Promise.resolve(true);
+    }
+
+    public async onEnd(): Promise<void> { 
+        await mongoose.disconnect();
+    }
+
+    public async onStart(): Promise<void> {
+        const config = ConfigManager.getInstance();
+        const itemSchema = new mongoose.Schema({
```

Comment:
Can we use the class defined in `entity` directory as schema? Define schema separately may cause problem for maintain schema.